docs/analyst_guide.md

```diff
@@ -0,0 +1,217 @@
+# Access levels
+
+Users can have one of the following access levels to a database:
+
+- `direct` - Direct (non-anonymized) access to data. Restrictions listed in this document do not apply in direct mode.
+- `anonymized_trusted` - Anonymized access to data. Prevents accidental release of personal data.
+- `anonymized_untrusted` - Anonymized access to data. Prevents intentional release of personal data.
+
+Use `SELECT diffix.access_level()` to see the current access level.
+
+# Anonymized queries
+
+In anonymized access level, queries targeting personal tables are restricted to a limited subset of SQL.
+A personal table is a relation that contains data of individuals or other protected entities.
+Administrators identify and mark such tables during configuration.
+
+When selecting data from personal tables, the following queries are allowed.
+
+## Queries with grouping
+
+Grouping queries have the following form:
+
+```
+SELECT col1, col2, ..., count(...)
+FROM personal_table
+GROUP BY col1, col2, ...
+```
+
+Where `col1`, `col2`, ... `colX` are table columns.
+Numeric and string columns can optionally be generalized.
+
+`count()` is any of the supported [count aggregate](#count) variants.
+
+**Example:**
+
+```
+SELECT city, last_name, count(*)
+FROM customers
+GROUP BY city, last_name
+```
+
+## Globally aggregating queries
+
+If no grouping columns are specified, the entire table is aggregated into a single bin.
+This bin is never suppressed, even if no rows exist in the table.
+
+```
+SELECT count(...), ...
+FROM personal_table
+```
+
+**Example:**
+
+```
+SELECT count(*), count(DISTINCT city)
+FROM customers
+```
+
+## Queries with implicit grouping
+
+The `GROUP BY` clause may be omitted:
+
+```
+SELECT col1, col2, ..., colX
+FROM personal_table
+```
+
+In this case, the results are grouped implicitly by the selected columns.
+Bins with insufficient individuals are filtered out, and the remaining
+bins are repeated for (anonymized) `count(*)` times.
+
+Semantically this is equivalent to:
+
+```
+SELECT col1, col2, ..., colX, generate_series(1, anon_count)
+FROM (
+  SELECT col1, col2, ..., colX, count(*) AS anon_count
+  FROM personal_table
+  GROUP BY col1, col2, ..., colX
+) x
+```
+
+**Example:**
+
+```
+SELECT city, last_name
+FROM customers
+```
+
+# Unrestricted queries
```

Comment:
I think this section is a bit confusing, because you are mixing the concepts of non-personal tables and post-processing. I think these two can be separated into two sections. (One of which simply says that tables are either personal or non-personal, and that the latter has no restrictions, and also show the command that tells whether a table is personal or non-personal)